pyrealm/core/solar.py

```diff
@@ -1,26 +1,584 @@
+
+    return _calc_daily_solar_radiation(rad_const, dr, ru, rv, k_pir, hs)
+
+
+def _calc_daily_solar_radiation(
+    rad_const: float, dr: NDArray, ru: NDArray, rv: NDArray, k_pir: float, hs: NDArray
+) -> NDArray:
+    """Calculate daily extraterrestrial solar radiation (J/m^2).
+
+    This function calculates the daily extraterrestrial solar radition (J/m^2)
+    using Eq. 1.10.3, Duffy & Beckman (1993)
+
+    Args:
+        rad_const   : planetary radiation constant, W/m^2
+        dr          : dimensionless distance factor
+        ru          : dimensionless variable substitute
+        rv          : dimensionless variable substitute
+        k_pir       : radians to degrees conversion, degrees/rad
+        hs          : local hour angle, degrees
+
+    Returns:
+        ra_d        : daily solar radiation, J/m^2
+    """
+
+    secs_d = 86400  # seconds in one earth day
+
+    ra_d = (
+        (secs_d / np.pi)
+        * rad_const
+        * dr
+        * (ru * k_pir * hs + rv * np.sin(np.deg2rad(hs)))
+    )
+
+    return ra_d
+
+
+def calc_transmissivity(k_c: float, k_d: float, sf: NDArray, elv: NDArray) -> NDArray:
+    r"""Calculate atmospheric transmissivity, tau.
+
+    This function calculates atmospheric transmissivity using the method of
+    Eq.11, Linacre (1968) and Eq 2, Allen (1996)
+
+    .. math::
+
+        \tau = (k\_c + k\_d \cdot sf) \cdot (1.0 + (2.67 \times 10^{-5}) \cdot elv)
+
+    Args:
+        k_c         : dimensionless cloudy transmissivity
+        k_d         : dimensionless angular coefficient of transmissivity
+        sf          : Daily sunshine fraction of observations, unitless
+        elv         : Elevation of observations, metres
+
+    Returns:
+        tau         : bulk transmissivity, unitless
+    """
+    tau = (k_c + k_d * sf) * (1.0 + (2.67e-5) * elv)
+    return tau
+
+
+def calc_ppfd_from_tau_ra_d(
+    k_fFEC: float, k_alb_vis: float, tau: NDArray, ra_d: NDArray
+) -> NDArray:
+    r"""Calculate photosynthetic photon flux density, PPFD,(mol/m^2).
+
+    This function calculates the PPFD in mol/m^2 from secondary calculated
+    variables and constants.
+
+    .. math::
+
+        ppfd = (1.0 \times 10^{-6}) \cdot k\_fFEC \cdot (1.0 - k\_alb\_vis) \cdot \tau \
+            cdot ra\_d
+
+    Args:
+        k_fFEC      : from flux to energy conversion, umol/J
+        k_alb_vis   : visible light albedo
+        tau         : bulk transmissivity, unitless
+        ra_d        : daily solar radiation, J/m^2
+
+    Returns:
+        ppfd        : photosynthetic photon flux density, mol/m^2
+    """
+
+    ppfd = (1.0e-6) * k_fFEC * (1.0 - k_alb_vis) * tau * ra_d
+
+    return ppfd
+
+
+def calc_ppfd(
+    sf: NDArray,
+    elv: NDArray,
+    latitude: NDArray,
+    julian_day: NDArray,
+    n_days: NDArray,
+    k_e: float,
+    k_eps: float,
+    k_pir: float,
+    k_d: float,
+    k_Gsc: float,
+    k_fFec: float,
+    k_alb_vis: float,
+    k_c: float,
```

Comment:
I'd be tempted here to use `const: CoreConst` just to make it easier to use and then pass the individual constants down to the sub-functions.